packages/@o3r/components/builders/metadata-check/index.it.spec.ts

```diff
@@ -0,0 +1,407 @@
+  const builderConfig = {
+    builder: '@o3r/components:check-config-migration-metadata',
+    options: builderOptions
+  };
+  angularJson.projects[projectName].architect['check-metadata'] = builderConfig;
+  writeFileAsJSON(angularJsonPath, angularJson);
+
+  // Add scope to project for registry management
+  let packageJson = existsSync(packageJsonPath) ? JSON.parse(readFileSync(packageJsonPath, { encoding: 'utf8' }).toString()) : undefined;
+  const packageName = `@o3r/${packageJson.name}-${packageNameSuffix}`;
+  packageJson = {
+    ...packageJson,
+    name: packageName,
+    private: false
+  };
+  writeFileAsJSON(packageJsonPath, packageJson);
+
+  // Set old metadata and publish to registry
+  writeFileAsJSON(metadataPath, previousConfigurationMetadata);
+
+  const bumpedVersion = inc(getLatestPackageVersion(packageName, execAppOptionsWorkspace) || baseVersion, 'patch');
+
+  const args = getPackageManager() === 'yarn' ? [] : ['--no-git-tag-version', '-f'];
+  packageManagerVersion(bumpedVersion, args, execAppOptions);
+  const userConfigPath = join(relative(projectPath, workspacePath), '.verdaccio/conf/.npmrc-logged');
+  packageManagerPublish(['--userconfig', userConfigPath], execAppOptions);
+
+  // Override with new metadata for comparison
+  writeFileAsJSON(metadataPath, newMetadata);
+
+  // Add migration data file
+  writeFileAsJSON(migrationDataPath, migrationData);
+};
+
+describe('check metadata migration', () => {
+  test('should not throw', () => {
+    initTest(
+      {
+        packageManager: undefined,
+        granularity: 'minor',
+        metadataPath: './component.config.metadata.json',
+        migrationDataPath: './apps/test-app/MIGRATION*.json',
+        allowBreakingChanges: true
+      },
+      newConfigurationMetadata,
+      defaultMigrationData,
+      'allow-breaking-changes'
+    );
+
+    const { workspacePath, projectName } = o3rEnvironment.testEnvironment;
+    const execAppOptionsWorkspace = { ...getDefaultExecSyncOptions(), cwd: workspacePath };
+
+    expect(() => packageManagerExec({ script: 'ng', args: ['run', `${projectName}:check-metadata`] }, execAppOptionsWorkspace)).not.toThrow();
+  });
+
+  test('should throw because no migration data', () => {
+    initTest(
+      {
+        granularity: 'minor',
+        metadataPath: './component.config.metadata.json',
+        migrationDataPath: './apps/test-app/MIGRATION*.json',
+        allowBreakingChanges: true,
+        packageManager: undefined
+      },
+      newConfigurationMetadata,
+      {
+        ...defaultMigrationData,
+        changes: []
+      },
+      'no-migration-data'
+    );
+    const { workspacePath, projectName } = o3rEnvironment.testEnvironment;
+    const execAppOptionsWorkspace = { ...getDefaultExecSyncOptions(), cwd: workspacePath };
+
+    try {
+      packageManagerExec({ script: 'ng', args: ['run', `${projectName}:check-metadata`] }, execAppOptionsWorkspace);
```

Comment:
this can probably be done using `expect(...).toThrow(expect.objectContaining({message: ...}))`

Reply:
as discussed it more complex to check several message and even more complex with the non presence of a message